### ISSUE TITLE
Domain is missing on routing field on MO. It should be limited to the one selected on bom

### ISSUE THREAD

**Issue by ybofr** [odoomrp/odoomrp-wip#515]
In the manufacturing order, there is no possibility to assign a BOM component to a specific operation. This is a big issue if  the material must be prepared just before the operation as it is the case for many companies.

**Comment by anajuaristi** [comment]
Yes. There is.
When and where did you try?

You can assign on planned materials before confirming one order
You can assign after confirming the order.

**Comment by anajuaristi** [comment]
Notice that routing must be assigned on BOM, otherwise system would consider standar manufacturing without operations, so, you are right there will be no possible to assign extra materials to one operation, simply because there is no operations on that production.

Try to just use a bom with routing.

**Comment by ybofr** [comment]
In this case, there is some confusion on the screen. I am allowed to choose any BOM and any routing, but in the case I choose the routing, the system does not consider it. This is not really understandable by the end user.
If the concept of Odoo is (wrongly imho) to assign a routing to a BOM, than you should not be allowed to do another assignment in the MO

I will test again with assigned routing.

**Comment by anajuaristi** [comment]
Yes. You are right a domain is missign on routing field in MO. It should be limited to selected bom.
Why do you say it's wrong concept assigning a routing to a bom? Could you please explain your point?
Thank you!!

**Comment by ybofr** [comment]
If you change the sequence of the operations, you will have a new routing, but not nessarily a new BOM. 
You also have the case Make or Buy for one operation of your routing. This does not require a change in the BOM.
And also, you may prepare different BOMs for the same routing (this is covered in standard)

**Comment by anajuaristi** [comment]
Conceptually this would be a huge change. 

I see something like you have got BOMs (materials to consume) and routings (operations to do), Mixing both many to many, (One bom several routes or one route several boms) would give you a kind of super-bom-route to be used on manufacturing.

It would be possible thinking about it in a new OdooMRP project phase but, as I said it would be really a big conceptual change.

**Comment by ybofr** [comment]
Yes, this is a huge change. I guess, the fact that you can easily duplicate a BOM in Odoo makes it less of a problem.
In other systems, they define manufacturing versions as an extra object that combines both. This way, you can choose the manufacturing version only and the system will select both routing and BOM.

The other point is that the manufacturing order is too late to assign the BOM components to the operations. On my last project, a SME, we had 30 operations and the user did not want to assign this manually when creating the manufacturing order as the people who create orders are not the one who design the routings and BOMs.

This is so far the biggest open point I found in the really great OpenMRP. It's really good and put Odoo on the map for manufacturing.

**Comment by anajuaristi** [comment]
Glad to hear you like it. :)
Just an small point. This version of our project now, is OdooMRP.
OpenMRP was our manufacturing improvements for OpenERP 6.1. It was hundreds of code lines inside a single module. You had to select everything or nothing. 

On the contrary, OdooMRP is totally modular. Almost any functionality can be installed or not depending on the customer needs so IMHO we did it much better now that we did before.
